packages/dart/noports_core/lib/src/common/host_validator.dart

```diff
@@ -0,0 +1,130 @@
+import 'dart:io';
+
+/// Utility class for validating and parsing host addresses
+class HostValidator {
```

Comment:
This is implemented, but most of it isn't being used in the rest of the PR, there's still plenty of other places interacting with the Dart:io APIs directly.